coops-ui/src/components/AnnualReport/AGMDate.vue

```diff
@@ -136,12 +136,10 @@ export default class AGMDate extends Mixins(DateMixin) {
    */
   private get maxDate (): string {
     /*
-     * If filing is for current year then use current date,
-     * otherwise use last day in filing year.
+     * If filing is in past year then use last day in that year,
+     * otherwise use current date.
      */
-    return (this.ARFilingYear === this.currentYear)
-      ? this.currentDate.split('/').join('-')
-      : `${this.ARFilingYear}-12-31`
+    return (this.ARFilingYear < this.currentYear) ? `${this.ARFilingYear}-12-31` : this.currentDate
```

Comment:
FYI - there are 2 "non changes" in this method:
1. currentDate did not have / so no need to split and rejoin
2. logic changed to < instead of === for better code safety